### PR TITLE
offchain/browser: allow for configuring the ctl-runtime

### PR DESCRIPTION
**Pull request by marijanp**
We want hello-world-browser to be able to connect to a configured cl-runtime. This PR introduces a way to configure the ctl-runtime, by reaing from a ctl-runtime-config.json.

other changes

- [ ] The impure tests via `nix run .#offchain:test` were run and all are passing.
- [ ] Where applicable, grammar, punctuation, and spelling within code and comments should be plausibly correct.
- [ ] The issue(s) that the work is addressing should be linked in the PR description.
- [ ] Any significant changes beyond what's clear from the linked issue(s) are documented in the PR description

### CODE REVIEW

File: offchain/hello-world-browser/src/Main.purs [hunk 1]
```diff
@@ -1,30 +1,81 @@
...
+                <> "Falling back to the default configuration."
+              warn $ show response
+              pure testnetConfig
+            Right ctlRuntimeConfig ->
+              pure $ testnetConfig
+                { ogmiosConfig = ctlRuntimeConfig.ogmiosConfig
+                , datumCacheConfig = ctlRuntimeConfig.datumCacheConfig
+                , ctlServerConfig = Just ctlRuntimeConfig.ctlServerConfig
+                }
+        StatusCode _ -> ctlRuntimeConfigFileNotFoundWarning
```
Comment (leomayleomay) [inline offchain/hello-world-browser/src/Main.purs:55]: could we log the non-200 status code?

File: offchain/hello-world-browser/flake-module.nix [hunk 2]
```diff
@@ -181,7 +186,10 @@
         "offchain:hello-world-browser:test" =
           offchain-lib.makeProjectShell hello-world-browser-test { };
       };
-      packages."offchain:hello-world-browser" = hello-world-browser.package;
+      packages = {
+        "offchain:hello-world-browser" = hello-world-browser.package;
+        "offchain:hello-world-browser-ctl-rt-config" = hello-world-browser.packageWithCtlRuntimeConfig { ogmiosConfig = { }; datumCacheConfig = { }; ctlServerConfig = { }; };
```
Comment (leomayleomay) [inline offchain/hello-world-browser/flake-module.nix:191]: maybe a new entry `offchain:hello-world-browser:serve-ctl-rt-config` in `apps` to host the package?

**Comment by leomayleomay** [comment]
hosting the `.#offchain:hello-world-browser` will error out since the ctl-rt-config.json will not be there

image

**Review by Geometer1729 — APPROVED** [review]
Asside from @leomayleomay comments LGTM

**Comment by marijanp** [comment]
> hosting the `.#offchain:hello-world-browser` will error out since the ctl-rt-config.json will not be there
> 
> image

It is a warning not an error

File: offchain/hello-world-browser/flake-module.nix (hunk 2 above)
Comment (marijanp) [reply offchain/hello-world-browser/flake-module.nix:191]: I can add this, but the usefulness is questionable. I added this package for testing, I will remove it in a successive PR. The `packageWithCtlRuntimeConfig` function was introduced such that I can be able to create a nixos-module for  `hello-world-browser`.

Comment (leomayleomay) [reply offchain/hello-world-browser/flake-module.nix:191]: ahh, I see, I thought this is going to be there running in parallel with `.#offchain:hello-world-browser`

**Review by leomayleomay — APPROVED** [review]
LGTM

File: offchain/hello-world-browser/src/Main.purs (hunk 1 above)
Comment (marijanp) [reply offchain/hello-world-browser/src/Main.purs]: I log the non-200 status codes too now

**Pull request merged**